tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,327 @@
+                "You can use the --timeout option to increase the timeout.%n", recordFetchTimeoutMs);
+        totalMessagesRead.set(messagesRead);
+        totalBytesRead.set(bytesRead);
+    }
+
+    protected static class ShareConsumerPerfOptions extends CommandDefaultOptions {
+        private final OptionSpec<String> brokerListOpt;
+        private final OptionSpec<String> bootstrapServerOpt;
+        private final OptionSpec<String> topicOpt;
+        private final OptionSpec<String> groupIdOpt;
+        private final OptionSpec<Integer> fetchSizeOpt;
+        private final OptionSpec<Void> resetBeginningOffsetOpt;
+        private final OptionSpec<Integer> socketBufferSizeOpt;
+        private final OptionSpec<Integer> numThreadsOpt;
+        private final OptionSpec<Integer> numFetchersOpt;
+        private final OptionSpec<String> consumerConfigOpt;
+        private final OptionSpec<Void> printMetricsOpt;
+        private final OptionSpec<Void> showDetailedStatsOpt;
+        private final OptionSpec<Long> recordFetchTimeoutOpt;
+        private final OptionSpec<Long> numMessagesOpt;
+        private final OptionSpec<Long> reportingIntervalOpt;
+        private final OptionSpec<String> dateFormatOpt;
+        private final OptionSpec<Void> hideHeaderOpt;
+
+        public ShareConsumerPerfOptions(String[] args) {
+            super(args);
+            brokerListOpt = parser.accepts("broker-list", "DEPRECATED, use --bootstrap-server instead; ignored if --bootstrap-server is specified. The broker list string in the form HOST1:PORT1,HOST2:PORT2.")
+                .withRequiredArg()
+                .describedAs("broker-list")
+                .ofType(String.class);
+            bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED unless --broker-list(deprecated) is specified. The server(s) to connect to.")
+                .requiredUnless("broker-list")
+                .withRequiredArg()
+                .describedAs("server to connect to")
+                .ofType(String.class);
+            topicOpt = parser.accepts("topic", "REQUIRED: The topic to consume from.")
+                .withRequiredArg()
+                .describedAs("topic")
+                .ofType(String.class);
+            groupIdOpt = parser.accepts("group", "The group id to consume on.")
+                .withRequiredArg()
+                .describedAs("gid")
+                .defaultsTo("perf-share-consumer-" + RND.nextInt(100_000))
+                .ofType(String.class);
+            fetchSizeOpt = parser.accepts("fetch-size", "The amount of data to fetch in a single request.")
+                .withRequiredArg()
+                .describedAs("size")
+                .ofType(Integer.class)
+                .defaultsTo(1024 * 1024);
+            resetBeginningOffsetOpt = parser.accepts("from-latest", "If the share consumer does not already have an established " +
```

Comment:
This will not work.

Reply:
agreed, we do not support --from-latest at this time., but I recall a conversation where we discussed that default case for share groups is to always start from the beginning of the log. However, we have plans in the future to give an option to the user to choose from beginning or latest. Hence, I added this option here.